src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs

```diff
@@ -190,19 +203,20 @@ internal class ClusterDnsEndPointDiscoverer : IEndPointDiscoverer {
 					} catch (Exception e) {
 						if (e is AggregateException ae)
 							e = ae.Flatten();
-						_log.Error("Failed to get cluster info from [{0}]: deserialization error: {1}.", endPoint, e);
+						_log.Error("Failed to get cluster info from [{0}]: request failed, error: {1}.",
+							endPoint.EndPoint, e);
+					} finally {
+						completed.Set();
 					}
 
 					completed.Set();
-				},
-				e => {
-					if (e is AggregateException ae)
+				}, e => {
+					if (e is AggregateException ae) 
 						e = ae.Flatten();
 					_log.Error("Failed to get cluster info from [{0}]: request failed, error: {1}.", endPoint, e);
```

Comment:
Based on previous changes, this should be `endPoint.EndPoint` in the log error message (it has not been changed on master too)